### PR TITLE
fix(tests/tox): updated gevent in tox config

### PR DESCRIPTION
**Pull request by avara1986**
## Description
Fixed gevent error in tox tests:
```
Collecting gevent<20.10,>=20.9
  Using cached gevent-20.9.0.tar.gz (5.8 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [60 lines of output]
      warning: src/gevent/resolver/cares.pyx:38:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/resolver/cares.pyx:40:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/resolver/cares.pyx:41:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      Compiling src/gevent/resolver/cares.pyx because it changed.
      [1/1] Cythonizing src/gevent/resolver/cares.pyx
      warning: src/gevent/libev/corecext.pyx:325:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:783:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:785:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:787:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: src/gevent/libev/corecext.pyx:791:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      Compiling src/gevent/libev/corecext.pyx because it changed.
      [1/1] Cythonizing src/gevent/libev/corecext.pyx
      Compiling src/gevent/_greenlet_primitives.py because it changed.
      [1/1] Cythonizing src/gevent/_greenlet_primitives.py
      Compiling src/gevent/_hub_primitives.py because it changed.
      [1/1] Cythonizing src/gevent/_hub_primitives.py
      Compiling src/gevent/_hub_local.py because it changed.
      [1/1] Cythonizing src/gevent/_hub_local.py
      Compiling src/gevent/_waiter.py because it changed.
      [1/1] Cythonizing src/gevent/_waiter.py
      warning: src/gevent/_gevent_cgreenlet.pxd:112:33: Declarations should not be declared inline.
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      cdef load_traceback
      cdef Waiter
      cdef wait
      cdef iwait
      cdef reraise
      cpdef GEVENT_CONFIG
            ^
      ------------------------------------------------------------
      
      src/gevent/_gevent_cgreenlet.pxd:181:6: Variables cannot be declared with 'cpdef'. Use 'cdef' instead.
      Compiling src/gevent/greenlet.py because it changed.
      [1/1] Cythonizing src/gevent/greenlet.py
      Traceback (most recent call last):
```

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -74,7 +74,7 @@ requires = virtualenv<=20.2.1
 
 [testenv]
 # Wheels for gevent segfault pretty easily
-install_command=python -m pip install --no-binary gevent {opts} {packages}
+install_command=python -m pip install Cython {opts} {packages}
```
Comment (brettlangdon) [inline tox.ini:77]: 🤔 Cython should be getting installed, unless the version of `pip` is really old? (<18 I think)

Comment (avara1986) [reply tox.ini:77]: are we testing old versions of pip? we should install Cython to install dd-trace-py package, isn't it? 

the correct command would be:
```
install_command=python -m pip install {opts} {packages}
```

Is that your suggestion? :+1:

Comment (brettlangdon) [reply tox.ini]: yes, that _should_ be the correct command. I'd be curious if it didn't work.

Comment (avara1986) [reply tox.ini]: I think the problem is that we should pass "--no-binary" to each "deps" installation, instead to install gevent globally in "install_command". WIP

**Review by juanjux — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Review by vdeturckheim — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 1.3

**Comment by mabdinur** [comment]
@Mergifyio backport 0.61

**Pull request merged**